File: vuln-sandbox/python/taint/xss-1hop.py
# Taint: XSS — 1-hop (Python)
# Source: request.args.get('name') → Sink: render_template_string
# Expected: TruePositive (taint/cwe79)
from flask import Flask, request, render_template_string

app = Flask(__name__)

@app.route('/greet')
def greet():
    name = request.args.get('name')            # SOURCE: HTTP request param
    return render_template_string(f"<h1>Hello {name}</h1>")  # SINK: template rendering

File: vuln-sandbox/node/taint/xss-1hop.js
// Taint: XSS — 1-hop (JS)
// Source: req.query.name → Sink: innerHTML
// Expected: TruePositive (taint/cwe79)
const express = require('express');
const app = express();

app.get('/greet', (req, res) => {
    const name = req.query.name;               // SOURCE: HTTP request param
    // SINK: innerHTML assignment with tainted data
    res.send(`<script>document.getElementById('msg').innerHTML = '${name}';</script>`);
});
